server/utils/viewUtils.ts

```diff
@@ -13,6 +13,14 @@ export default class ViewUtils {
     return escape(val).val
   }
 
+  static nl2br(val: string): string {
+    // handle things like nunjucks macros (which are 'SafeStrings') that do not need formatting
+    if (typeof val !== 'string') return val
+
+    const nl2br = new nunjucks.Environment().getFilter('nl2br')
+    return nl2br(ViewUtils.escape(val))
```

Comment:
Again, would be good to have a test that describes the behaviour (i.e. that it escapes).

Reply:
Thinking about it some more, I don't think that this method should be doing any escaping. I think we want to be able to look at any of our view code and be able to tell, at a glance, that it's correctly escaping values. We can currently do that since all of the escaping is done directly in the view code. With this change, developers will have to start carrying around the additional piece of knowledge that if they see a `ViewUtils.nl2br` then that's also correctly escaping things too.

Reply:
no drama - will remove.